screenpy/configuration.py

```diff
@@ -24,58 +28,84 @@
     import tomli as tomllib
 
 
-# The following pyproject.toml function was taken and adapted from Black:
+# The following pyproject.toml functions were taken and adapted from Black:
 # https://github.com/psf/black/blob/main/src/black/files.py
 
 
-def _parse_pyproject_toml(tool_path: str) -> dict[str, Any]:
-    """Parse a pyproject toml file, pulling out relevant parts for ScreenPy.
+class PyprojectTomlConfig(PydanticBaseSettingsSource):
+    """Load setting configuration from pyproject.toml."""
```

Comment:
This was the biggest change.

Pydantic moved its settings management stuff into `pydantic-settings`, and changed a whole heckin' lot over there.

Some of the changes actually made things a lot easier for us, 'cause now you can just inherit from `ScreenPySettings` when you make your own configuration in an extension. Neat!